k9mail/src/main/java/com/fsck/k9/activity/setup/AccountSetupPresenter.java

```diff
@@ -243,14 +280,89 @@ protected ProviderInfo doInBackground(Void... params) {
                 if (providerInfo != null) return providerInfo;
 
                 providerInfo = autodiscover.findProviderInfo(email);
+                if (providerInfo != null) return providerInfo;
 
-                return providerInfo;
+                testDomain(domain);
+
+                return null;
+            }
+
+            private void testDomain(String domain) {
```

Comment:
I found it's not so convenient to make these a class because it should modify the `accountConfig` directly(rather than return a `ProviderInfo` object like other `Autoconfigure*` classes do) and modify some global variables(`incomingReady` and `outgoingReady`), what's more some methods are in module `k9mail` but not `k9mail-library`(though it can be simply moved to `k9mail-library` if we want). I think it may be better to do other things first and then try to spend time on it.